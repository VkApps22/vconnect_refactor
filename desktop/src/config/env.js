const HOST_ADDRESS =
  process.env.REACT_APP_HOST_ADDRESS || 'http://localhost:3000';

const NATIVE_REDIRECT_LINK =
  process.env.REACT_APP_NATIVE_REDIRECT_LINK || 'vconnect://authorize';

export default {
  HOST_ADDRESS,
  NATIVE_REDIRECT_LINK,
};
