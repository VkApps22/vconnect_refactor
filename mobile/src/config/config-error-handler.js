import axios from 'axios';
import ServiceError from '../services/service-error';

export default (onNetworkError) => {
  const axiosErrorHandler = async (error) => {
    if (error.response) {
      if (
        error.response.status === 401 ||
        error.response.status === 403 ||
        error.response.status === 404
      )
        onNetworkError();

      throw new ServiceError(
        error.response.data && error.response.data.messageCode
          ? error.response.data.messageCode
          : 'aw-snap-something-went-wrong',
        error
      );
    } else if (error.request) {
      onNetworkError();
      throw new ServiceError('server-cannot-be-reached', error);
    } else {
      throw error;
    }
  };

  axios.interceptors.response.use(
    (response) => response,
    (error) => axiosErrorHandler(error)
  );
};
