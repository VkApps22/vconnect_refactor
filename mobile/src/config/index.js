export { default as env } from './env';
export { default as configErrorHandler } from './config-error-handler';
