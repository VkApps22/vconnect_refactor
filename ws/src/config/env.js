const NODE_ENV = process.env.NODE_ENV || 'local';

const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost/vconnect';
const JWT_SHARED_KEY =
  process.env.JWT_SHARED_KEY ||
  '70337336763979244226452948404D635166546A576E5A7134743777217A25432A462D4A614E645267556B58703273357538782F413F4428472B4B6250655368';

const FB_APP_ID = process.env.FB_APP_ID || '1067192480465819';
const FB_APP_SECRET = process.env.FB_APP_SECRET || '';

const LN_APP_ID = process.env.LN_APP_ID || '779zvhm4srbatp';
const LN_APP_SECRET = process.env.LN_APP_SECRET || '';

const SMTP_HOST = process.env.SMTP_HOST || 'smtp.gmail.com';
const SMTP_PORT = process.env.SMTP_PORT || 465;
const SMTP_SECURE = process.env.SMTP_SECURE || true;
const SMTP_USERNAME =
  process.env.SMTP_USERNAME || 'devtestexperiment2020@gmail.com';
const SMTP_PASSWORD = process.env.SMTP_PASSWORD || 'cCK9zTdFzb';

const EMAIL_FROM = process.env.EMAIL_FROM || 'devtestexperiment2020@gmail.com';
const EMAIL_FROM_NAME = process.env.EMAIL_FROM_NAME || 'VConnect â€“ Platform';

const COMPANY_EMAIL_DOMAIN = process.env.COMPANY_EMAIL_DOMAIN || 'vulkan.com';

const APPLE_APP_ID = process.env.APPLE_APP_ID || 'com.vulkan.vconnect';
const APPLE_APP_SECRET = process.env.APPLE_APP_SECRET || '669D5WPNT8';
const APPLE_TEAM_ID = process.env.APPLE_TEAM_ID || 'WTBN98QD3K';
const APPLE_PRIVATE_KEY =
  process.env.APPLE_PRIVATE_KEY ||
  '-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgQoh5lZWChj2H97j8\ntAvgOQHpuKMlirosZa9lJlbrjKygCgYIKoZIzj0DAQehRANCAAROTjyhA5s+q7e9\nIGuX79c0ix+sg39L90Yp3vXrRS9eUyj9s7YG8MRyuhSJwCK3g3VpFgvsT49EuE1o\n477zcmlj\n-----END PRIVATE KEY-----';

export default {
  NODE_ENV,

  MONGODB_URI,
  JWT_SHARED_KEY,

  FB_APP_ID,
  FB_APP_SECRET,

  LN_APP_ID,
  LN_APP_SECRET,

  SMTP_HOST,
  SMTP_PORT,
  SMTP_SECURE,
  SMTP_USERNAME,
  SMTP_PASSWORD,

  EMAIL_FROM,
  EMAIL_FROM_NAME,

  COMPANY_EMAIL_DOMAIN,

  APPLE_APP_ID,
  APPLE_APP_SECRET,
  APPLE_TEAM_ID,
  APPLE_PRIVATE_KEY,
};
