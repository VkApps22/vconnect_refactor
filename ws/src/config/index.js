export { default as env } from './env';
export { default as mongo } from './mongo';
