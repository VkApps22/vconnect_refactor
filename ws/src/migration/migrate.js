import { Migration } from '../domain/entity';

export default async () => {
  const migrations = await Migration.find({ appliedAt: null })
    .sort({ createdAt: 1 })
    .lean();

  const promises = migrations.map((migration) => import(`./${migration.name}`));
  const imports = await Promise.all(promises);

  const execute = async (migration, index) => {
    try {
      const func = imports[index].default;
      await func();

      await Migration.updateOne(
        { name: migration.name },
        {
          appliedAt: new Date(),
          error: undefined,
        }
      );
    } catch (err) {
      await Migration.updateOne(
        { name: migration.name },
        {
          error: err.message ? err.message.toString() : err.toString(),
        }
      );
    }
  };

  for (let i = 0; i < migrations.length; i += 1) {
    // eslint-disable-next-line no-await-in-loop
    await execute(migrations[i], i);
  }
};
