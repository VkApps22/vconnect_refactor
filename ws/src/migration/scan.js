import fs from 'fs';
import { Migration } from '../domain/entity';

const getDirectories = async (source) => {
  const directories = await fs.promises.readdir(source, {
    withFileTypes: true,
  });
  return directories
    .filter((entry) => entry.isDirectory())
    .filter((entry) => entry.name.match(/\d+-.+/gm) !== null);
};

export default async () => {
  const available = await getDirectories(__dirname);
  const existing = await Migration.find({}).lean();

  const newMigrations = available.filter(
    (directory) =>
      !existing.some((migration) => migration.name === directory.name)
  );

  const getTimestamp = (name) => parseInt(name.split('-')[0], 10);

  const promises = newMigrations.map((migration) =>
    Migration.create({
      name: migration.name,
      createdAt: new Date(getTimestamp(migration.name)),
      detectedAt: new Date(),
      appliedAt: null,
    })
  );

  await Promise.all(promises);
};
