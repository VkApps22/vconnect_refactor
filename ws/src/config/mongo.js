import mongoose from 'mongoose';
import env from './env';

const options = {
  useNewUrlParser: true,
  useUnifiedTopology: true,
  useCreateIndex: true,
  useFindAndModify: false,
};

const mongo = {
  connect: async () => {
    await mongoose.connect(global.__MONGO_URI__ || env.MONGODB_URI, options);
    mongoose.Schema.Types.String.checkRequired((v) => typeof v === 'string');

    process.on('SIGINT', () => {
      mongoose.connection.close(() => {
        process.exit(0);
      });
    });

    mongoose.connection.on('error', (err) => {
      console.error(`Mongoose! connection error: ${err}`);
    });
  },
  disconnect: async () => {
    await mongoose.disconnect();
  },
  clear: async () => {
    const { collections } = mongoose.connection;

    const results = [];
    Object.keys(collections).forEach((key) => {
      if (Object.prototype.hasOwnProperty.call(collections, key))
        results.push(collections[key].deleteMany());
    });

    await Promise.all(results);
  },
};

export default mongo;
