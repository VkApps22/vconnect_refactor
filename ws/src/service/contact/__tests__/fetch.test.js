import fetchContactService from '../fetch';
import { mongo } from '../../../config';
import scan from '../../../migration/scan';
import migrate from '../../../migration/migrate';

describe('fetchContact', () => {
  beforeAll(async () => {
    await mongo.connect();
    await scan();
    await migrate();
  });
  afterAll(() => mongo.disconnect());

  const getTestSignature = (contacts) =>
    contacts.map(
      (contact) =>
        `${contact.language}_${contact.categoryKey}_${contact.placeKey}_${contact.name}`
    );

  it('happy day', async () => {
    const contacts = await fetchContactService({});

    expect(contacts.length).toBe(15);
    expect(getTestSignature(contacts)).toMatchSnapshot();
  });

  it('country is germany and state is unknown', async () => {
    const contacts = await fetchContactService({
      country: 'germany',
    });

    expect(contacts.length).toBe(15);
    expect(getTestSignature(contacts)).toMatchSnapshot();
  });

  it('country is brazil and state is acre', async () => {
    const contacts = await fetchContactService({
      country: 'brazil',
      state: 'ac',
    });

    expect(contacts.length).toBe(15);
    expect(getTestSignature(contacts)).toMatchSnapshot();
  });

  it('country is brazil and state is sp', async () => {
    const contacts = await fetchContactService({
      country: 'brazil',
      state: 'sp',
    });

    expect(contacts.length).toBe(15);
    expect(getTestSignature(contacts)).toMatchSnapshot();
  });

  it('country is brazil and state is sc', async () => {
    const contacts = await fetchContactService({
      country: 'brazil',
      state: 'sc',
    });

    expect(contacts.length).toBe(15);
    expect(getTestSignature(contacts)).toMatchSnapshot();
  });
});
