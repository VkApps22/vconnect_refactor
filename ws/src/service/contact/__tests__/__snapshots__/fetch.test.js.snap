// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchContact country is brazil and state is acre 1`] = `
Array [
  "en_customer-service_undefined_Ednei Leite",
  "en_sales_undefined_Brazilian Plant",
  "es_customer-service_undefined_Ednei Leite",
  "es_sales_demas-paises-de-america-latina_Arno Ett",
  "es_sales_sucursal-colombia-y-region-andina_Andre Nunes",
  "es_sales_sucursal-mexico_Gerardo Duran",
  "pt_customer-service_undefined_Ednei Leite",
  "pt_sales_para-regiao-norte_Alison Pinheiro",
  "pt_sales_minas-gerais_Guilherme Lajes",
  "pt_sales_minas-gerais_Luiz Reimão Nogueira",
  "pt_sales_minas-gerais-e-maranhao_Rogério Monticelli",
  "pt_sales_parana-santa-catarina-e-rio-grande-do-sul_Leandro Coelho",
  "pt_sales_rio-de-janeiro-e-vale-do-paraiba-sp_Manoel Bueno",
  "pt_sales_sao-paulo-capital_Matias Quaglia",
  "pt_sales_sao-paulo-interior_Daniel Cunha",
]
`;

exports[`fetchContact country is brazil and state is sc 1`] = `
Array [
  "en_customer-service_undefined_Ednei Leite",
  "en_sales_undefined_Brazilian Plant",
  "es_customer-service_undefined_Ednei Leite",
  "es_sales_demas-paises-de-america-latina_Arno Ett",
  "es_sales_sucursal-colombia-y-region-andina_Andre Nunes",
  "es_sales_sucursal-mexico_Gerardo Duran",
  "pt_customer-service_undefined_Ednei Leite",
  "pt_sales_parana-santa-catarina-e-rio-grande-do-sul_Leandro Coelho",
  "pt_sales_minas-gerais_Guilherme Lajes",
  "pt_sales_minas-gerais_Luiz Reimão Nogueira",
  "pt_sales_minas-gerais-e-maranhao_Rogério Monticelli",
  "pt_sales_para-regiao-norte_Alison Pinheiro",
  "pt_sales_rio-de-janeiro-e-vale-do-paraiba-sp_Manoel Bueno",
  "pt_sales_sao-paulo-capital_Matias Quaglia",
  "pt_sales_sao-paulo-interior_Daniel Cunha",
]
`;

exports[`fetchContact country is brazil and state is sp 1`] = `
Array [
  "en_customer-service_undefined_Ednei Leite",
  "en_sales_undefined_Brazilian Plant",
  "es_customer-service_undefined_Ednei Leite",
  "es_sales_demas-paises-de-america-latina_Arno Ett",
  "es_sales_sucursal-colombia-y-region-andina_Andre Nunes",
  "es_sales_sucursal-mexico_Gerardo Duran",
  "pt_customer-service_undefined_Ednei Leite",
  "pt_sales_sao-paulo-capital_Matias Quaglia",
  "pt_sales_sao-paulo-interior_Daniel Cunha",
  "pt_sales_rio-de-janeiro-e-vale-do-paraiba-sp_Manoel Bueno",
  "pt_sales_minas-gerais_Guilherme Lajes",
  "pt_sales_minas-gerais_Luiz Reimão Nogueira",
  "pt_sales_minas-gerais-e-maranhao_Rogério Monticelli",
  "pt_sales_para-regiao-norte_Alison Pinheiro",
  "pt_sales_parana-santa-catarina-e-rio-grande-do-sul_Leandro Coelho",
]
`;

exports[`fetchContact country is germany and state is unknown 1`] = `
Array [
  "en_customer-service_undefined_Ednei Leite",
  "en_sales_undefined_Brazilian Plant",
  "es_customer-service_undefined_Ednei Leite",
  "es_sales_demas-paises-de-america-latina_Arno Ett",
  "es_sales_sucursal-colombia-y-region-andina_Andre Nunes",
  "es_sales_sucursal-mexico_Gerardo Duran",
  "pt_customer-service_undefined_Ednei Leite",
  "pt_sales_minas-gerais_Guilherme Lajes",
  "pt_sales_minas-gerais_Luiz Reimão Nogueira",
  "pt_sales_minas-gerais-e-maranhao_Rogério Monticelli",
  "pt_sales_para-regiao-norte_Alison Pinheiro",
  "pt_sales_parana-santa-catarina-e-rio-grande-do-sul_Leandro Coelho",
  "pt_sales_rio-de-janeiro-e-vale-do-paraiba-sp_Manoel Bueno",
  "pt_sales_sao-paulo-capital_Matias Quaglia",
  "pt_sales_sao-paulo-interior_Daniel Cunha",
]
`;

exports[`fetchContact happy day 1`] = `
Array [
  "en_customer-service_undefined_Ednei Leite",
  "en_sales_undefined_Brazilian Plant",
  "es_customer-service_undefined_Ednei Leite",
  "es_sales_demas-paises-de-america-latina_Arno Ett",
  "es_sales_sucursal-colombia-y-region-andina_Andre Nunes",
  "es_sales_sucursal-mexico_Gerardo Duran",
  "pt_customer-service_undefined_Ednei Leite",
  "pt_sales_minas-gerais_Guilherme Lajes",
  "pt_sales_minas-gerais_Luiz Reimão Nogueira",
  "pt_sales_minas-gerais-e-maranhao_Rogério Monticelli",
  "pt_sales_para-regiao-norte_Alison Pinheiro",
  "pt_sales_parana-santa-catarina-e-rio-grande-do-sul_Leandro Coelho",
  "pt_sales_rio-de-janeiro-e-vale-do-paraiba-sp_Manoel Bueno",
  "pt_sales_sao-paulo-capital_Matias Quaglia",
  "pt_sales_sao-paulo-interior_Daniel Cunha",
]
`;
